src/sa_pool.c

```diff
@@ -506,31 +506,29 @@ int sa_fetch(struct netif_port *dev, const struct sockaddr_in *daddr,
     int err;
     assert(saddr);
 
-    if (saddr->sin_addr.s_addr != INADDR_ANY && saddr->sin_port != 0)
+    if (saddr &&  saddr->sin_addr.s_addr && saddr->sin_addr.s_addr != INADDR_ANY && saddr->sin_port != 0)
```

Comment:
1. `INADDR_ANY` is zero, so  `saddr->sin_addr.s_addr` is not needed.
2. actually `if (saddr->sin_addr.s_addr != INADDR_ANY && saddr->sin_port != 0)` is OK, because here's an `assert(saddr)` has already make sure `saddr` is not NULL.